magento/pwa-studio
# Update Home Page content after cache hit (#2230)

## Description

Apollo will not query the GraphQL endpoint for a new version of CMS Page content due to the usage of the default fetch policy, cache only. This PR updates this cache policy to use `cache-and-network` (https://www.apollographql.com/docs/react/api/react-apollo/#optionsfetchpolicy) which will ensure the cached data is rendered, then a request is made to retrieve any new data if available.

This fetch policy ensures a quick user experience while ensuring up to date content is rendered on the homepage. If there was a change in content between the cache and GraphQL Apollo automatically handles re-rendering this.

## Related Issue
Closes #2230 .
https://jira.corp.magento.com/browse/PWA-471

## Acceptance 



### Verification Stakeholders
@jimbo 
@sirugh 
@brendanfalkowski 
@dhaecker 

### Specification


https://www.apollographql.com/docs/react/api/react-apollo/#optionsfetchpolicy

### Verification Steps
- Navigate to homepage in app (allowing content to be cached)
- Open Admin
- Go to Content > Pages > Home page
- Edit the home page and save
- Refresh the application (the old content will display, which is expected for the _first_ reload)
- Refresh the application again
- The old content will be displayed again, as it will for any number of times that the page is refreshed
- Manually clear the localStorage property, {{apollo-cache-persist}}
- Refresh the application
- _Now_ the updated home page data will be displayed

## Screenshots / Screen Captures (if appropriate)
![2020-06-02 16 24 40](https://user-images.githubusercontent.com/1774374/83571618-ba6afe80-a4ed-11ea-81f0-8e497c222b0a.gif)

**New Experience:**
![2020-06-12 10 06 55](https://user-images.githubusercontent.com/1774374/84517175-8386ac80-ac94-11ea-9f09-4c7ae577961d.gif)

## Checklist
